sky/sdk/lib/editing/input.dart

```diff
@@ -64,10 +56,12 @@ class Input extends StatefulComponent {
 
   Widget build() {
     ThemeData themeData = Theme.of(this);
+    bool focused = Focus.at(this);
 
-    if (focused && !_isAttachedToKeyboard) {
-      keyboard.show(_editableValue.stub);
-      _isAttachedToKeyboard = true;
+    if (focused && !_keyboardHandle.attached) {
```

Comment:
Yeah, this confused me in the previous version too.  The subtle thing is that if you query Focus.at, then the way you get notified when the focus state changes is that you get scheduled to rebuild.  That means the build() function is effectively the focus and blur event handler.